control-plane/connect-inject/peering_acceptor_controller.go

```diff
@@ -52,16 +54,32 @@ func (r *PeeringAcceptorController) Reconcile(ctx context.Context, req ctrl.Requ
 	// If the PeeringAcceptor resource has been deleted (and we get an IsNotFound
 	// error), we need to delete it in Consul.
 	if k8serrors.IsNotFound(err) {
-		r.Log.Info("PeeringAcceptor was deleted, deleting from Consul", "name", req.Name, "ns", req.Namespace)
-		if err := r.deletePeering(ctx, req.Name); err != nil {
-			return ctrl.Result{}, err
-		}
+		r.Log.Info("PeeringAcceptor resource not found. Ignoring.", "name", req.Name, "ns", req.Namespace)
 		return ctrl.Result{}, nil
 	} else if err != nil {
 		r.Log.Error(err, "failed to get PeeringAcceptor", "name", req.Name, "ns", req.Namespace)
 		return ctrl.Result{}, err
 	}
 
+	if acceptor.GetDeletionTimestamp().IsZero() {
+		if !controllerutil.ContainsFinalizer(acceptor, FinalizerName) {
+			controllerutil.AddFinalizer(acceptor, FinalizerName)
+			if err := r.Update(ctx, acceptor); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		if containsString(acceptor.Finalizers, FinalizerName) {
+			r.Log.Info("PeeringAcceptor was deleted, deleting from Consul", "name", req.Name, "ns", req.Namespace)
+			if err := r.deletePeering(ctx, req.Name); err != nil {
+				return ctrl.Result{}, err
+			}
+			controllerutil.RemoveFinalizer(acceptor, FinalizerName)
+			err := r.Update(ctx, acceptor)
+			return ctrl.Result{}, err
```

Comment:
After returning at this point, does the Acceptor resource still exist, just without a finalizer? And then K8s will delete the Acceptor automatically?

Reply:
yup. at this point, the acceptor resource will exist without the finalizer and get immediately GC-ed by kubernetes.